federation/pkg/federation-controller/service/servicecontroller.go

```
@@ -728,8 +728,9 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
-				ensuredCount += 1
+				if err := s.ensureDnsRecords(clusterName, service); err == nil {
+					ensuredCount += 1
+				}
```

Should we return the error directly ?